### PR TITLE
Tali awesome new feature with cool boxes initially

### PR DESCRIPTION
**Pull request by taliGav**

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -28,11 +28,33 @@
...
-@keyframes App-logo-spin {
+.App-main {
+  background-color: yellow;
+  min-height: 90vh;
+  display: inline-flexbox;
+  flex-direction: column;
+  justify-content:center
+}
+
+.Json-input-container {
```
Comment (naorzr) [inline src/App.css]: avoid using classes for components for now.
use styles inline

File: src/App.css [hunk 2]
```diff
@@ -28,11 +28,33 @@
...
+  background-color: green;
+  /* display: inline-flexbox; */
+  /* display: flex; */
+  max-width: 40%;
+}
+
+.Ts-output-container {
```
Comment (naorzr) [inline src/App.css]: avoid using classes, put styles inline

File: src/cmps/Json-input.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+function JsonInput() {
+    return (
+        <div className="Json-input-container">
+            JSON input main container
+
+            <form action="" method="post">
+                JSON input form container
+
+
+                {/* <textarea name="Json-input-text" id="" cols="30" rows="10"></textarea> */}
```
Comment (naorzr) [inline src/cmps/Json-input.tsx]: remove comment and spaces

File: src/cmps/Typescript-output.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+
+
+
+
+function TypeScriptOutput() {
+    return (
+        <div className="Ts-output-container">
+
+
```
Comment (naorzr) [inline src/cmps/Typescript-output.tsx]: fix spaces

File: src/try.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+//         "id": 0,
+//         "name": "Blanche Mathews",
+//         "age":"20",
+//         "isSmiling": true
+//     },
+//     {
+//         "id": 1,
+//         "name": "Salinas Bruce",
+//         "isSmiling": false
+//     },
```
Comment (naorzr) [inline src/try.js:26]: remove comments

File: src/try.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+
+// console.log(...x);
+
+// var y = JSON.stringify(...x,)
+
+// var y = JSON.stringify(x).split(",")
+
+
+function typesOfValues(x) {
```
Comment (naorzr) [inline src/try.js:42]: whats the point of this function?
how is it any good, if I can't map the values back to the keys?

File: src/try.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    return Object.values(x).map(x => typeof (x))
+}
+
+function tsInterfaceOutput(x) {
+    var keys = Object.keys(x)
```
Comment (naorzr) [inline src/try.js]: dont use var.
keys doesn't change, use const

File: src/try.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    var types = typesOfValues(x)
+    
+    let entArrs = []
```
Comment (naorzr) [inline src/try.js]: no need to use let

File: src/try.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    for (let i = 0; i < keys.length; i++) {
+        let arr = []
```
Comment (naorzr) [inline src/try.js]: no need to use let

File: src/try.js [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+        arr[0] = keys[i];
+        arr[1] = types[i]
+
+        entArrs.push(arr)
+    }
+
+    let entries = new Map(entArrs)
```
Comment (naorzr) [inline src/try.js]: no need to put it in a map to use Object.fromEntries()

File: src/try.js [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+function typesOfValues(x) {
+    return Object.values(x).map(x => typeof (x))
+}
+
+function tsInterfaceOutput(x) {
+    var keys = Object.keys(x)
+    var types = typesOfValues(x)
+    
+    let entArrs = []
+    for (let i = 0; i < keys.length; i++) {
```
Comment (naorzr) [inline src/try.js]: u can use a map on entries to achieve a more elegant solution instead of all this

**Review by naorzr — CHANGES_REQUESTED** [review]

File: src/App.css (hunk 1 above)
Comment (taliGav) [reply src/App.css]: 👍

File: src/App.css (hunk 2 above)
Comment (taliGav) [reply src/App.css]: 👍

File: src/cmps/Typescript-output.tsx (hunk 4 above)
Comment (taliGav) [reply src/cmps/Typescript-output.tsx]: OK motek. fixed

File: src/cmps/Json-input.tsx (hunk 3 above)
Comment (taliGav) [reply src/cmps/Json-input.tsx]: comment was left intentionally for future use

File: src/try.js (hunk 6 above)
Comment (taliGav) [reply src/try.js:42]: function was written for tests. I'll leave it here for now ❤

File: src/try.js (hunk 5 above)
Comment (taliGav) [reply src/try.js:26]: an example of another json for future use

File: src/try.js (hunk 8 above)
Comment (taliGav) [reply src/try.js]: ok

File: src/try.js (hunk 9 above)
Comment (taliGav) [reply src/try.js]: ok

File: src/try.js (hunk 10 above)
Comment (taliGav) [reply src/try.js]: ok

File: src/try.js (hunk 11 above)
Comment (taliGav) [reply src/try.js]: ok. done

File: src/App.css (hunk 2 above)
Comment (taliGav) [reply src/App.css]: still ugly
styles updated to inline 😘

**Pull request closed without merging**